packages/ai-jsx/src/core/completion.tsx

```diff
@@ -14,6 +15,18 @@ export type ModelPropsWithChildren = ModelProps & {
 
 export type ModelComponent<T extends ModelPropsWithChildren> = Component<T>;
 
+export interface FunctionDefinition {
+  name: string;
+  description?: string;
+  parameters: Record<string, FunctionParameter>;
+}
+
+export interface FunctionParameter {
+  description?: string;
+  type?: string;
+  required: boolean;
+}
```

Comment:
Hmm, the OpenAI module doesn't export these types?

Reply:
OpenAI exposes a type (`ChatCompletionFunctions`) that is a bit incomplete (doesn't explicitly mark all settable fields for parameters; instead uses a Record<string, any>), and is optimized for API data transfer (there are some "type": "object" that need to be set on that Record<string, any> that is meaningless in terms of Function interface definition). My goal here is to decouple the AI.JSX function definition from the underlying API implementation (which may be OpenAI or otherwise).